sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2024-2025.bpmn

```diff
@@ -789,92 +839,144 @@ dmnX (comes from dmn)</bpmn:text>
+      <bpmn:extensionElements>
+        <zeebe:script expression="=if(calculatedDataAdditionalTransportationAllowance &#62; 0)&#10;then&#10;max(inputDataMinimumTotalTransportationAllowance,inputDataEstimatedTotalTransportAllowance)&#10;else&#10;  inputDataMinimumTotalTransportationAllowance" resultVariable="calculatedDataTotalTransportationAllowance" />
+        <zeebe:ioMapping>
+          <zeebe:input source="=// Minimum transportation allowance eligible for an onsite offering.&#10;// This allowance is calculated as the allowance that a student will receive for an onsite offering with no additional transportation involved.&#10;dmnPartTimeProgramYearMaximums.limitTransportationAllowance * offeringWeeks" target="inputDataMinimumTotalTransportationAllowance" />
+          <zeebe:input source="=(offeringWeeks - calculatedDataAdditionalTransportWeeks) * dmnPartTimeProgramYearMaximums.limitTransportationAllowance + calculatedDataAdditionalTransportationAllowance" target="inputDataEstimatedTotalTransportAllowance" />
+        </zeebe:ioMapping>
+      </bpmn:extensionElements>
+      <bpmn:incoming>Flow_049nq15</bpmn:incoming>
+      <bpmn:outgoing>Flow_0iugroz</bpmn:outgoing>
     </bpmn:scriptTask>
-    <bpmn:scriptTask id="Activity_1vwijyc" name="Define Dependents">
+    <bpmn:intermediateThrowEvent id="Event_07s4lhn" name="Calculate additional transportation allowance">
       <bpmn:extensionElements>
-        <zeebe:script expression="=// Consider only dependents who were born on or before the study end date.&#10;inputDataDependents[date(item.dateOfBirth) &#60;= date(offeringStudyEndDate)]" resultVariable="calculatedDataDependants" />
         <zeebe:ioMapping>
-          <zeebe:input source="=//TODO: Student appeal variable needs to be considered here.&#10;if studentDataDependants = null &#10;then []&#10;else studentDataDependants" target="inputDataDependents" />
+          <zeebe:output source="=calculatedDataAdditionalTransportationWeeklyCost * calculatedDataAdditionalTransportWeeks" target="calculatedDataAdditionalTransportationAllowance" />
         </zeebe:ioMapping>
       </bpmn:extensionElements>
-      <bpmn:incoming>Flow_13hzh8q</bpmn:incoming>
-      <bpmn:outgoing>Flow_0xi2yp3</bpmn:outgoing>
+      <bpmn:incoming>Flow_08k878p</bpmn:incoming>
+      <bpmn:outgoing>Flow_1i635oj</bpmn:outgoing>
+    </bpmn:intermediateThrowEvent>
+    <bpmn:intermediateThrowEvent id="Event_15d88qb" name="Calculate additional transportation allowance">
+      <bpmn:extensionElements>
+        <zeebe:ioMapping>
+          <zeebe:output source="=0" target="calculatedDataAdditionalTransportationAllowance" />
+        </zeebe:ioMapping>
+      </bpmn:extensionElements>
+      <bpmn:incoming>Flow_0x59561</bpmn:incoming>
+      <bpmn:outgoing>Flow_17p7duv</bpmn:outgoing>
+    </bpmn:intermediateThrowEvent>
+    <bpmn:scriptTask id="Activity_1sdvd13" name="Calculate additional transportation weekly cost">
```

Comment:
Not a blocker at all, with the idea of using the script tasks and because the input variables belong only to the scope of the task, I believe that as much as possible the FEEL expression should be as short as possible.
For instance, for the below
```
// If the student is receiving weekly transportation allowance from the institution for clinical or practicum placement
// then their limit for additional transportation weekly allowance must be reduced.
if (calculatedDataAdditionalTransportPlacement = true) then
  // The maximim weekly additional transportation cost cannot be greater than student entered weekly cost.
  min((dmnPartTimeProgramYearMaximums.limitAdditionalTransportationAllowance - dmnPartTimeProgramYearMaximums.limitAdditionalTransportationPlacement), calculatedDataAdditionalTransportCost)
else
  // The maximim weekly additional transportation cost cannot be greater than student entered weekly cost.
  min(dmnPartTimeProgramYearMaximums.limitAdditionalTransportationAllowance, calculatedDataAdditionalTransportCost)
```
I would not see any harm in pre-calculating the below to improve the readability, event being used just in one block of the if condition.
```
dmnPartTimeProgramYearMaximums.limitAdditionalTransportationAllowance - dmnPartTimeProgramYearMaximums.limitAdditionalTransportationPlacement
```

Reply:
I love the idea. It is more atomic.